crates/sui-graphql-rpc/src/types/transaction_block_effects.rs

```diff
@@ -35,21 +36,12 @@ pub enum ExecutionStatus {
 #[Object]
 impl TransactionBlockEffects {
     /// The transaction that ran to produce these effects.
-    async fn transaction_block(&self, ctx: &Context<'_>) -> Result<TransactionBlock> {
-        if let Some(stored_tx) = &self.stored {
-            Ok(TransactionBlock::try_from(stored_tx.clone()).extend()?)
-        } else {
-            let digest = self.native.transaction_digest().to_string();
-            ctx.data_unchecked::<PgManager>()
-                .fetch_tx(&digest)
-                .await
-                .transpose()
-                .ok_or(Error::Internal(format!(
-                    "Unable to fetch known transaction with digest {}",
-                    digest
-                )))?
-                .extend()
-        }
+    async fn transaction_block(&self) -> Result<Option<TransactionBlock>> {
```

Comment:
You should be able to create a `transaction_block` with the `right` side of the `tx_data` as well -- is that coming in a follow-up PR?

Reply:
Yes